MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -1312,21 +1311,24 @@
 }
 
 #define addMTLVertexFormatDesc(MTL_VTX_FMT, IOS_CAPS, MACOS_CAPS)  \
-	MVKAssert(fmtIdx < _mtlVertexFormatCount, "Attempting to describe %d MTLVertexFormats, but only have space for %d. Increase the value of _mtlVertexFormatCount", fmtIdx + 1, _mtlVertexFormatCount);  \
-	_mtlVertexFormatDescriptions[fmtIdx++] = { .mtlVertexFormat = MTLVertexFormat ##MTL_VTX_FMT, VK_FORMAT_UNDEFINED,  \
-                                               mvkSelectPlatformValue<MVKMTLFmtCaps>(kMVKMTLFmtCaps ##MACOS_CAPS, kMVKMTLFmtCaps ##IOS_CAPS),  \
-                                               MVKMTLViewClass::None, MTLPixelFormatInvalid, "MTLVertexFormat" #MTL_VTX_FMT }
+	mtlVtxFmt = MTLVertexFormat ##MTL_VTX_FMT;  \
+	if (mtlVtxFmt < _mtlVertexFormatCount) {  \
+		_mtlVertexFormatDescriptions[mtlVtxFmt] = { .mtlVertexFormat = mtlVtxFmt, VK_FORMAT_UNDEFINED,  \
+													mvkSelectPlatformValue<MVKMTLFmtCaps>(kMVKMTLFmtCaps ##MACOS_CAPS, kMVKMTLFmtCaps ##IOS_CAPS),  \
+													MVKMTLViewClass::None, MTLPixelFormatInvalid, "MTLVertexFormat" #MTL_VTX_FMT };  \
+	} else {  \
+		MVKAssert(false, "Attempting to describe at least %lu MTLVertexFormats, but only have space for %d. Increase the value of _mtlVertexFormatCount", mtlVtxFmt + 1, _mtlVertexFormatCount);  \
```

Comment:
`assert(false)` is an antipattern.

Reply:
Yeah. I know. I couldn't think of another way of doing this in a way that would cover both Debug and Release builds, without placing the same test in both the `if()` and assertion tests.

Any suggestions for a replacement?

Reply:
This is supposed to be an internal precondition that must not be violated, no matter what. I would assume that developers working on that part would be using `Debug` builds while working on it. The `if` thus shouldn't be necessary in `Release` builds.

In fact, this really ought to be a `static_assert()`, so it gets caught at compile time instead of at run time.

Reply:
Fixed. Redesign has eliminated a need for assertions.